modules/oai-ran-cu-k8s/variables.tf

```diff
@@ -0,0 +1,88 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+variable "model_name" {
+  description = "Name of Juju model to deploy applications to."
+  type        = string
+  default     = "ran-cu"
+}
+
+variable "create_model" {
+  description = "Allows to skip Juju model creation and re-use a model created in a higher level module."
+  type        = bool
+  default     = true
+}
+
+variable "cu_channel" {
+  description = "The channel to use when deploying Charmed OAI RAN CU."
+  type        = string
+  default     = "2.1/edge"
+}
+variable "cu_config" {
+  description = "Application config for the CU. Details about available options can be found at https://charmhub.io/oai-ran-cu-k8s/configurations."
+  type        = map(string)
+  default     = {}
+}
+
+# Grafana Agent K8s
+
+variable "grafana_agent_channel" {
+  description = "The channel to use when deploying `grafana-agent-k8s` charm."
+  type        = string
+  default     = "latest/stable"
+}
+variable "grafana_agent_config" {
+  description = "Additional configuration for the Grafana Agent. Details about available options can be found at https://charmhub.io/grafana-agent-k8s/configure."
+  type        = map(string)
+  default     = {}
+}
+
+# Canonical Observability Stack (COS)
+
+variable "deploy_cos" {
+  description = "When set to `true`, COS will be deployed along with SD-Core. COS will use a separate model on the same Juju controller."
+  type        = bool
+  default     = false
+}
+variable "use_existing_cos" {
+  description = "Allows integrating with an existing COS deployment. When set to `true`, `cos_model_name`, `prometheus_remote_write_offer_url` and `loki_logging_offer_url` are required."
+  type        = bool
+  default     = false
+}
+
+variable "cos_model_name" {
+  description = "Name of COS Juju model. This model will be created if `deploy_cos` is set to true or used as a source when integrating with existing COS deployment."
+  type        = string
+  default     = "cos-lite"
+}
```

Comment:
I think we should not deploy COS inside this module and simply support an existing COS.

I am fine to keep it this way until there is an official COS solution module.

Reply:
I removed it right away. I've spent some more time thinking about this and I didn't find a good use case for having it here.